develop/interoperability/xcm-runtime-apis.md

```diff
@@ -0,0 +1,360 @@
+    ```
+
+    ***Output***
+
+    --8<-- 'code/develop/interoperability/xcm-runtime-apis/query-assets-output.html'
+
+    ---
+
+### Query XCM Weight
+
+Calculates the weight required to execute a given XCM message. It is useful for estimating the execution cost of a cross-chain message in the destination chain before sending it.
+
+```rust
+--8<-- 'https://raw.githubusercontent.com/paritytech/polkadot-sdk/refs/heads/stable2412/polkadot/xcm/xcm-runtime-apis/src/fees.rs:50:50'
+```
+
+??? interface "Input parameters"
+
+    `message` ++"VersionedXcm<()>"++ <span class="required" markdown>++"required"++</span>
+    
+    A versioned XCM message whose execution weight is being queried.
+
+    ---
+
+??? interface "Output parameters"
+
+    ++"Result<Weight, Error>"++
+    
+    The calculated weight required to execute the provided XCM message. If the calculation fails, an error is returned instead.
+
+    ---
+
+??? interface "Example"
+
+    This example demonstrates how to calculate the weight needed to execute a teleport transfer from the Paseo network to the Paseo Asset Hub parachain using the XCM Payment API. The result shows the required weight in terms of reference time and proof size needed in the destination chain.
+
+    ***Usage with PAPI***
+
+    ```js
+    --8<-- 'code/develop/interoperability/xcm-runtime-apis/query-xcm-weight.js'
+    ```
+
+    !!!note
+        Make sure to replace `INSERT_USER_ADDRESS` with your SS58 address before running the script.
+
+    ***Output***
+
+    --8<-- 'code/develop/interoperability/xcm-runtime-apis/query-xcm-weight-output.html'
+
+    ---
+
+### Query Weight to Asset Fee
+
+Converts a given weight into the corresponding fee for a specified `AssetId`. It allows clients to determine the cost of execution in terms of the desired asset.
+
+```rust
+--8<-- 'https://raw.githubusercontent.com/paritytech/polkadot-sdk/refs/heads/stable2412/polkadot/xcm/xcm-runtime-apis/src/fees.rs:58:58'
+```
+
+??? interface "Input parameters"
+
+    `weight` ++"Weight"++ <span class="required" markdown>++"required"++</span>
+    
+    The execution weight to be converted into a fee.
+
+    ---
+
+    `asset` ++"VersionedAssetId"++ <span class="required" markdown>++"required"++</span>
+    
+    The asset in which the fee will be calculated. This must be a versioned asset ID compatible with the runtime.
+
+    ---
+
+??? interface "Output parameters"
+
+    ++"Result<u128, Error>"++
+    
+    The fee needed to pay for the execution for the given `AssetId.`
+
+    ---
+
+??? interface "Example"
+
+    This example demonstrates how to calculate the fee for a given execution weight using a specific versioned asset ID (PAS token) on Paseo Asset Hub.
+
+    ***Usage with PAPI***
+
+    ```js
+    --8<-- 'code/develop/interoperability/xcm-runtime-apis/query-weight-to-fee.js'
+    ```
+
+    ***Output***
+
+    --8<-- 'code/develop/interoperability/xcm-runtime-apis/query-weight-to-fee-output.html'
+    ---
```

Comment:
I would definitely add an example after this that brings it all together.

We've made the error before of having "docs" be just a reference. That doesn't really tell the user how to use our tools.

Reply:
We will include a step-by-step explanation on a tutorial explaining how to use them together